src/vector/backends/awkward_constructors.py

```diff
@@ -256,11 +271,20 @@ def Array(*args: typing.Any, **kwargs: typing.Any) -> typing.Any:
     import vector
     import vector.backends.awkward  # noqa: 401
 
+    if isinstance(args[0], dict):
+        if all(isinstance(x, list) for x in args[0].values()):
+            raise TypeError(
+                "Use vector.zip to construct vectors using a list of coordinates"
+            )
```

Comment:
See above: this iterates over the data in Python. Besides not being general (it's only catching the case of dicts of lists, not array inputs, and not even non-list iterables), we only want to iterate over the data once.

To make a point about checking type metadata versus iterating over the data, checking each instance, consider this:

```python
In [1]: import awkward._v2 as ak   # I'm doing this in v2 for the ak.from_parquet function

In [2]: some_data = ak.from_parquet("https://pivarski-princeton.s3.amazonaws.com
   ...: /chep-2021-jagged-jagged-jagged/zlib9-jagged1.parquet", row_groups=[0])

In [3]: %%timeit
   ...: isinstance(ak.type(some_data.layout), ak.types.ListType)
   ...: 
   ...: 
104 µs ± 430 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

In [4]: some_data[0]
Out[4]: <Array [-0.423, 2.34, -0.757, ..., -0.297, 0.597, -0.298] type='11 * float32'>

In [5]: isinstance(some_data[0], list)
Out[5]: False

In [6]: from collections.abc import Iterable

In [7]: isinstance(some_data[0], Iterable)
Out[7]: True

In [8]: %%timeit
   ...: all(isinstance(x, Iterable) for x in some_data)
   ...: 
   ...: 
9.98 s ± 48.7 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [9]: len(some_data)
Out[9]: 1342176
```

It's a reasonably large array, just over a million entries, but you can check the type metadata almost _a million times faster_ than iterating over the data: 104 µs versus 9.98 s. (Note: if you're given builtin types like lists and dicts, the iteration is faster, such that checking type metadata is merely a thousand times faster than iterating over lists and dicts: 161 ms. But still, _a thousand times faster_.)

Of course, you might be _given_ lists and dicts, and then what can you do? You have to eventually turn it into an array, anyway (see [ak.to_layout](https://awkward-array.readthedocs.io/en/latest/_auto/ak.to_layout.html) above), but even doing that is only 2.5 seconds for a million entries. (The process of building the array involves type-checks in every entry.)

```python
In [10]: as_lists = ak.to_list(some_data)

In [11]: %%timeit
    ...: as_array_again = ak.from_iter(as_lists)
    ...: 
    ...: 
2.51 s ± 28.1 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Bottom line: you'll eventually have an array with type metadata. The data in the array will not be like Python objects, with unknown type. Use the type metadata!